### PR TITLE
[OTX][Hot-Fix] Fix self-sl seg issue

### PR DESCRIPTION
**Pull request by sungmanc**
# Summary
For self-supervised learning, there is no argument `--val-data-roots`.
In this case, `seg/trainer.py` need to check the whether `val` data is included in data configuration to avoid making `val_dataloader`

Here is the local test result for Self-SL segmentation
image

### CODE REVIEW

**Review by cih9088 — APPROVED** [review]

**Review by cih9088 — DISMISSED** [review]
Could you apply the change in det/trainer.py as well for completeness?

**Review by goodsong81 — DISMISSED** [review]

**Comment by sungmanc** [comment]
> Could you apply the change in seg/trainer.py as well for completeness?

No, problems, I'll

**Comment by sungmanc** [comment]
@goodsong81 @cih9088 , I applied this solution to `det/train.py` to consider further self-sl detection.
Please recheck this :)

File: otx/mpa/det/trainer.py [hunk 1]
```diff
@@ -56,7 +56,8 @@ def run(self, model_cfg, model_ckpt, data_cfg, **kwargs):
         datasets = [build_dataset(cfg.data.train)]
 
         # FIXME: Currently detection do not support multi batch evaluation. This will be fixed
-        cfg.data.val_dataloader.samples_per_gpu = 1
+        if 'val' in cfg.data:
+            cfg.data.val_dataloader.samples_per_gpu = 1
 
         # FIXME: scale_factors is fixed at 1 even batch_size > 1 in simple_test_mask
         # Need to investigate, possibly due to OpenVINO
```
Comment (cih9088) [inline otx/mpa/det/trainer.py]: You missed this block (line 62-66)! Please include it as well. :)

Comment (sungmanc) [reply otx/mpa/det/trainer.py]: I'm going to remove line 62-66 because if there is a validation dataset, `val_dataloader.samplers_per_gpu` will be set to 1.

Comment (sungmanc) [reply otx/mpa/det/trainer.py]: What do you think?

Comment (cih9088) [reply otx/mpa/det/trainer.py]: Yes, then no objection on that. I also left this part for the reference because there is a `FIXME` I didn't consider at all.
If this `FIXME` is considered `FIXED`, then I believe we can safely get rid of it.

**Review by cih9088 — DISMISSED** [review]

File: otx/mpa/det/trainer.py (hunk 1 above)
Comment (sungmanc) [reply otx/mpa/det/trainer.py]: Actually, in the previous version, there is no problem to use `val_dataloader.samplers_per_gpu` > 1 for detection, classification except for instance segmentation, semantic segmentation. However, I'm also not sure the latest mmdetection version. So, let's keep as is with `FIXME` tag.

File: otx/mpa/seg/trainer.py [hunk 2]
```diff
@@ -57,7 +57,9 @@ def run(self, model_cfg, model_ckpt, data_cfg, **kwargs):
         datasets = [build_dataset(cfg.data.train)]
 
         # FIXME: Currently segmentor does not support multi batch evaluation.
-        cfg.data.val_dataloader.samples_per_gpu = 1
+        # For the Self-SL case, there is no val data. So, need to check the 
+        if 'val' in cfg.data:
+            cfg.data.val_dataloader.samples_per_gpu = 1
```
Comment (sungchul1) [inline otx/mpa/seg/trainer.py:62]: Could you explain why this change is needed for seg? I don't know the reason because mmseg already uses `samples_per_gpu=1` when creating val_dataloader. 

https://github.com/openvinotoolkit/mmsegmentation/blob/ac8e04d73857af3bff4c27df62ac69d1c51ff3ac/mmseg/apis/train.py#L65-L74

Comment (cih9088) [reply otx/mpa/seg/trainer.py:62]: **This is a new feature from public mm libraries to construct different dataloader per subset.**

Since global options for dataloader, such as `samples_per_gpu`, etc., would be deprecated in the future as `mmdet` is on the way, MPA explicitly [construct dataloader config](https://github.com/openvinotoolkit/training_extensions/blob/8b1732239eeedde534f1f36e7d89d8951c4d5bb3/otx/mpa/seg/trainer.py#L104) for each subset right before the training. [The method](https://github.com/openvinotoolkit/training_extensions/blob/8b1732239eeedde534f1f36e7d89d8951c4d5bb3/otx/mpa/stage.py#L309-L346) just pops the global options and assigns them to each dataloader config.

Specifying dataloader config would [override the default](https://github.com/open-mmlab/mmsegmentation/blob/master/mmseg/apis/train.py#L157-L162) which is 1 for validations's `samples_per_gpu`, MPA should set to `1` explicitly.

Comment (sungmanc) [reply otx/mpa/seg/trainer.py:62]: My changing part is to avoid the `cfg.data.val_dataloader.samples_per_gpu=1`.
And after checking the mmseg, I also think we can remove the line #60-62 for the segmentation.

Comment (sungmanc) [reply otx/mpa/seg/trainer.py:62]: > Specifying dataloader config would [override the default](https://github.com/open-mmlab/mmsegmentation/blob/master/mmseg/apis/train.py#L157-L162) which is 1 for validations's samples_per_gpu, MPA should set to 1 explicitly.

Then you mean that we need to explicitly let users know by adding this line although the default value is set to 1?

Comment (cih9088) [reply otx/mpa/seg/trainer.py:62]: @sungmanc This should be as is. Unless, it will build dataloader with `samples_per_gpu` from global dataloader option which might not be `1`.

Comment (sungmanc) [reply otx/mpa/seg/trainer.py:62]: Understood, thanks for kind explanation. what do you think? @sungchul2 

If you also agree, we can merge this PR

Comment (sungchul1) [reply otx/mpa/seg/trainer.py:62]: I agree. Thanks for your kind summary!

**Review by sungchul1 — DISMISSED** [review]

**Review by cih9088 — DISMISSED** [review]

**Comment by sungmanc** [comment]
@cih9088 @sungchul2 @goodsong81 , after rebase, need to re-approve.  thanks in advance.

**Review by cih9088 — APPROVED** [review]

**Review by sungchul1 — APPROVED** [review]
